### DISCUSSION TITLE
Running PGLite in a Chrome Extension Background Script

### DISCUSSION THREAD

**Discussion by talperetz** [electric-sql/pglite#237] [Q&A]
First off, I want to say I really appreciate the work you've done with pglite—it's a fantastic tool!

I'm currently trying to integrate pglite into a Chrome extension background script. I've attempted using both PGLiteWorker and PGlite, but I haven't been successful in getting it to work.

Could you provide any guidance or a working example for this use case? Or is it currently not supported?

Thanks in advance for your help!

**Comment by samwillis** [comment]
Hey @talperetz 

I don't have any experience with Chrome Extensions and so can't offer much advice, but I'm very happy to give pointers and suggestions. Do you have any errors or example code I can take a look at?

The `PGLiteWorker` if fairly geared towards a multi-tab browser environment and so may not be the best abstraction.

**Reply by talperetz** [reply]
I'm getting `Uncaught ReferenceError: XMLHttpRequest is not defined` 

In a Chrome extension's service worker, XMLHttpRequest is not available because service workers are designed to work with the Fetch API instead.

**Reply by samwillis** [reply]
Ah, so we use Emscripten to compile PGlite. It emits JS code that uses XMLHTTPRequest to retrieve data files, that's probably the code path that it's throwing on. We would need to make significant patches to Emscrioten to replace that with Fetch, and so I suspect that is unlikely in the short/medium term.

I would look at if there are any XMLHTTPRequest polyfills (wrapping Fetch) that you can use in a Chrime extension.

**Reply by pmp-p** [reply]
for reference https://github.com/emscripten-core/emscripten/issues/6241#issuecomment-1259498993
